### ISSUE TITLE
[Image Request]: Kubernetes Pause

### ISSUE THREAD

**Issue by stevehipwell** [chainguard-images/images#219]
### What software would you like us to add to our image catalog? (one per issue please)

[Kubernetes Pause](https://github.com/kubernetes/kubernetes/blob/master/build/pause/Dockerfile).

### which versions of the containerized software do you need?

Latest

### Add some justification for why this specific image and versions are important.

The official image isn't signed, but if the official image was signed this might not be needed.

**Comment by amouat** [comment]
This is interesting! We'll look into getting the official image signed (it should be) but also a chainguard image might make sense...

**Comment by cpanato** [comment]
pause image is getting signed for `>3.8`

```
$ cosign verify registry.k8s.io/pause:3.9

Verification for registry.k8s.io/pause:3.9 --
The following checks were performed on each of these signatures:
  - The cosign claims were validated
  - Existence of the claims in the transparency log was verified offline
  - Any certificates were verified against the Fulcio roots.

[{"critical":{"identity":{"docker-reference":"europe-west4-docker.pkg.dev/k8s-artifacts-prod/images/pause"},"image":{"docker-manifest-digest":"sha256:7031c1b283388d2c2e09b57badb803c05ebed362dc88d84b480cc47f72a21097"},"type":"cosign container image signature"},"optional":{"1.3.6.1.4.1.57264.1.1":"https://accounts.google.com","Bundle":{"SignedEntryTimestamp":"MEQCIG9k5P8vM60/p0lb1STULPjs71DtF1e7QtDZJWV63XF/AiAcCmRRzg9yZznCm6/+wv0WbQF2SE0VUA1EYDTLiG87Rw==","Payload":{"body":"eyJhcGlWZXJzaW9uIjoiMC4wLjEiLCJraW5kIjoiaGFzaGVkcmVrb3JkIiwic3BlYyI6eyJkYXRhIjp7Imhhc2giOnsiYWxnb3JpdGhtIjoic2hhMjU2IiwidmFsdWUiOiIyZjNhOGU1OGM0YjhiYmFlMThkYzc0YWY0ZTVjNzFkYmQ3OWIxNzJmNGE2YjFjZGNmM2Q5YTRiZjE1N2E2ZWJjIn19LCJzaWduYXR1cmUiOnsiY29udGVudCI6Ik1FUUNJQVowLzRlRFRzQmx3a3QrQjFJN21aWjRVcjlaRHlrc24waDVpVXhlK05FckFpQmZqL3ZrbWtzendJQWxzenB1VTNRTTh1VktrM3Z4ZWhaV3hDN3ZLVi9GRWc9PSIsInB1YmxpY0tleSI6eyJjb250ZW50IjoiTFMwdExTMUNSVWRKVGlCRFJWSlVTVVpKUTBGVVJTMHRMUzB0Q2sxSlNVTjZha05EUVd4UFowRjNTVUpCWjBsVlJHVlpWMkZTU3pKaVpFMXViRlJNVFRCRGEwbGFXSFpMUml0VmQwTm5XVWxMYjFwSmVtb3dSVUYzVFhjS1RucEZWazFDVFVkQk1WVkZRMmhOVFdNeWJHNWpNMUoyWTIxVmRWcEhWakpOVWpSM1NFRlpSRlpSVVVSRmVGWjZZVmRrZW1SSE9YbGFVekZ3WW01U2JBcGpiVEZzV2tkc2FHUkhWWGRJYUdOT1RXcEplRTFVUlRCTlZHZDVUbXBSTlZkb1kwNU5ha2w0VFZSRk1FMVVaM3BPYWxFMVYycEJRVTFHYTNkRmQxbElDa3R2V2tsNmFqQkRRVkZaU1V0dldrbDZhakJFUVZGalJGRm5RVVUyVlhGblUxb3pXVWd4UjBkRFpFOW5SV05GTlhsWUwybFdVVVJ0UWt0emJURmljVzRLVjFwQ1drMVpNbnB2YXpsVE9XMUNOMDB6Vlc1VmVuVmpOaXMzUm1SVU9VTjFTRk5EY2t4SU1ISktNaXRRUWtORlRtRlBRMEZZU1hkblowWjFUVUUwUndwQk1WVmtSSGRGUWk5M1VVVkJkMGxJWjBSQlZFSm5UbFpJVTFWRlJFUkJTMEpuWjNKQ1owVkdRbEZqUkVGNlFXUkNaMDVXU0ZFMFJVWm5VVlZVVG1OM0NraG9RamhHYjJneVpqTm5iMHhPWkZSeGFEVlJhVXRWZDBoM1dVUldVakJxUWtKbmQwWnZRVlV6T1ZCd2VqRlphMFZhWWpWeFRtcHdTMFpYYVhocE5Ga0tXa1E0ZDFSM1dVUldVakJTUVZGSUwwSkZWWGRSTkVaQ1lYcG9la3hYYkhWYWJrcG9URmRrYW1OcE1YZGpiVGwwWWpOU2JHTnJRbkpQU0UxMFdWaEtNQXBoVjFwb1dUTlNla3hZUW5saU1sRjFZVmRHZEV4dFpIcGFXRW95WVZkT2JGbFhUbXBpTTFaMVpFTTFhbUl5TUhkTFVWbExTM2RaUWtKQlIwUjJla0ZDQ2tGUlVXSmhTRkl3WTBoTk5reDVPV2haTWs1MlpGYzFNR041Tlc1aU1qbHVZa2RWZFZreU9YUk5TVWRMUW1kdmNrSm5SVVZCWkZvMVFXZFJRMEpJZDBVS1pXZENORUZJV1VFelZEQjNZWE5pU0VWVVNtcEhValJqYlZkak0wRnhTa3RZY21wbFVFc3pMMmcwY0hsblF6aHdOMjgwUVVGQlIwVmtNbFJrY1VGQlFRcENRVTFCVW5wQ1JrRnBRazlQVW5aMlRUSTRNSFFyTUdwclN6UXdTR3BtVjNkM2F5dDRla2xtU1hoRVRVTkdkMjV1Vm10alNHZEphRUZPWWxoc1RrNTFDa3BPTjA0elNqZHBTWE12WlRNMVQwZElaM0Z5UVZCRk5ucHNhSEJUYVRKQldraHVjMDFCYjBkRFEzRkhVMDAwT1VKQlRVUkJNbXRCVFVkWlEwMVJRMHdLUTFCUFNqSjFTWFZJUVZZM1lVTTFiMmMwY0dVM1YzTlZWRFJuU0dSblUyeDFhazl5VlVsdmIwc3hRV05vZEdjMWFqaHBTemxGYlc4emMyWnVOMk13UXdwTlVVTlRaMUYwUm5SRk5UVmxLMEV2ZWxkcU4yVjNZWGhrUjJkR05qQlhiMXBpZG5oMVdFaEljVVowWVZFNFdFWkhhMVJxY21OMVpUSXpVazF3TlcxS0NtWjBZejBLTFMwdExTMUZUa1FnUTBWU1ZFbEdTVU5CVkVVdExTMHRMUW89In19fX0=","integratedTime":1668450411,"logIndex":7073269,"logID":"c0d23d6ad406973f9559f3ba2d1ca01f84147d8ffc5b8445c224f98b9591801d"}},"Issuer":"https://accounts.google.com","Subject":"k8s-infra-gcr-promoter@k8s-artifacts-prod.iam.gserviceaccount.com","org.kubernetes.kpromo.mirrors":"europe-west4-docker.pkg.dev/k8s-artifacts-prod/images/pause,asia-east1-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/pause,asia-south1-docker.pkg.dev/k8s-artifacts-prod/images/pause,southamerica-west1-docker.pkg.dev/k8s-artifacts-prod/images/pause,europe-west9-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/pause,europe-southwest1-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/pause,us-south1-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/pause,southamerica-west1-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/pause,asia.gcr.io/k8s-artifacts-prod/pause,europe-west2-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/pause,australia-southeast1-docker.pkg.dev/k8s-artifacts-prod/images/pause,europe-north1-docker.pkg.dev/k8s-artifacts-prod/images/pause,us-east1-docker.pkg.dev/k8s-artifacts-prod/images/pause,us-east1-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/pause,us-west1-docker.pkg.dev/k8s-artifacts-prod/images/pause,eu.gcr.io/k8s-artifacts-prod/kubernetes/pause,us.gcr.io/k8s-artifacts-prod/pause,asia-northeast2-docker.pkg.dev/k8s-artifacts-prod/images/pause,us-west1-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/pause,us-south1-docker.pkg.dev/k8s-artifacts-prod/images/pause,eu.gcr.io/k8s-artifacts-prod/pause,europe-west8-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/pause,australia-southeast1-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/pause,us-central1-docker.pkg.dev/k8s-artifacts-prod/images/pause,us-west2-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/pause,us.gcr.io/k8s-artifacts-prod/kubernetes/pause,us-east5-docker.pkg.dev/k8s-artifacts-prod/images/pause,europe-west1-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/pause,us-east5-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/pause,europe-southwest1-docker.pkg.dev/k8s-artifacts-prod/images/pause,europe-west2-docker.pkg.dev/k8s-artifacts-prod/images/pause,asia-northeast1-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/pause,europe-north1-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/pause,asia-northeast1-docker.pkg.dev/k8s-artifacts-prod/images/pause,asia.gcr.io/k8s-artifacts-prod/kubernetes/pause,asia-northeast2-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/pause,asia-east1-docker.pkg.dev/k8s-artifacts-prod/images/pause,us-east4-docker.pkg.dev/k8s-artifacts-prod/images/pause,europe-west4-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/pause,europe-west1-docker.pkg.dev/k8s-artifacts-prod/images/pause,us-west2-docker.pkg.dev/k8s-artifacts-prod/images/pause,asia-south1-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/pause,us-central1-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/pause,europe-west8-docker.pkg.dev/k8s-artifacts-prod/images/pause,us-east4-docker.pkg.dev/k8s-artifacts-prod/images/kubernetes/pause,europe-west9-docker.pkg.dev/k8s-artifacts-prod/images/pause"}}]
```

**Comment by amouat** [comment]
We had a look. As @cpanato mentioned, new versions of the image are signed. But a lot of distros like kops are still using older versions which are unsigned.

There's not really anything in this image and I think it's on top of scratch, so I don't think it's worth creating a Chainguard Image atm. Please let me know if you'd still find one useful for any reason and we can definitely do it!

Closing this for the minute.

**Comment by stevehipwell** [comment]
The specific tags are signed (I missed that) but `latest` isn't.

```shell
~ > cosign verify registry.k8s.io/pause:latest
Error: no matching signatures:

main.go:62: error during command execution: no matching signatures:
```

**Comment by amouat** [comment]
`latest` is 8 years old and from k8s 1.0. Definitely don't use that image. I think @cpanato is going to look into promoting a newer version to the same tag.

**Comment by stevehipwell** [comment]
Thanks for looking into this @amouat, upstream Kubernetes does seem like the correct place to fix this. I'm using `3.9` even though it feels wrong to use a non-SemVer version tag. I had spotted that `latest` wasn't a good tag to use after I checked the metadata, but a lot of the examples including [Cluster Autoscaler overprovisioning](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#how-can-i-configure-overprovisioning-with-cluster-autoscaler) use the `latest` tag (by not using a tag in the CA case).
